### PR TITLE
feat: add functionality to create cohort distribution function

### PR DESCRIPTION
**Pull request by andrewmorris43**
## Description

This PR implements 3 of the 4 functions that should be called from the `CreateCohortDistribution` function and adds unit tests. These functions are as follows:

- Get participant data (this is just a stub for now whilst it is still being implemented)
- Allocate service provider
- Transform participant data
- Add cohort distribution

## Context

[DTOSS-3223](https://nhsd-jira.digital.nhs.uk/browse/DTOSS-3223)

## Type of changes

- [x] Refactoring (non-breaking change)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [x] I have followed the code style of the project
- [x] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [x] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: application/CohortManager/src/Functions/CohortDistributionServices/CreateCohortDistribution/CreateCohortDistribution.cs [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+        bool error = false;
+        CohortDistributionParticipant participantData;
+        string serviceProvider;
+        CohortDistributionParticipant transformedParticipant = new CohortDistributionParticipant();
+
+        // Get Participant Data
+        // this is a stub for a function that doesn't yet exist
+        // it just returns a hardcoded participant for the time being
+        try
```
Comment (MWClayson-NHS) [inline application/CohortManager/src/Functions/CohortDistributionServices/CreateCohortDistribution/CreateCohortDistribution.cs:63]: Please change it in to an api call that points to a stub.

File: application/CohortManager/src/Functions/CohortDistributionServices/CreateCohortDistribution/CreateCohortDistribution.cs
Comment (MWClayson-NHS) [inline application/CohortManager/src/Functions/CohortDistributionServices/CreateCohortDistribution/CreateCohortDistribution.cs:1]: This is a quite a long function which makes things hard to read, please consider breaking down parts of it into individual methods.

File: application/CohortManager/src/Functions/CohortDistributionServices/CreateCohortDistribution/CreateCohortDistribution.cs [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+                _logger.LogInformation("Called transform data service");
+
+                using (StreamReader reader = new StreamReader(response.GetResponseStream(), Encoding.UTF8))
+                {
+                    var body = await reader.ReadToEndAsync();
+                    CohortDistributionParticipant result = JsonSerializer.Deserialize<CohortDistributionParticipant>(body);
+                    transformedParticipant = result;
+                }
+            }
+            else error = true;
```
Comment (MWClayson-NHS) [inline application/CohortManager/src/Functions/CohortDistributionServices/CreateCohortDistribution/CreateCohortDistribution.cs:136]: I think you should just return a badrequest it we hit these error = true's as i think we should stop the processing if we get an error response from any of them.

**Review by MWClayson-NHS — CHANGES_REQUESTED** [review]
Please review my suggested changes, Happy to discuss over a call if you need any clarification.

File: application/CohortManager/src/Functions/CohortDistributionServices/CreateCohortDistribution/CreateCohortDistribution.cs (hunk 2 above)
Comment (andrewmorris43) [reply application/CohortManager/src/Functions/CohortDistributionServices/CreateCohortDistribution/CreateCohortDistribution.cs]: I've made this change thanks.

File: application/CohortManager/src/Functions/CohortDistributionServices/CreateCohortDistribution/CreateCohortDistribution.cs (hunk 1 above)
Comment (andrewmorris43) [reply application/CohortManager/src/Functions/CohortDistributionServices/CreateCohortDistribution/CreateCohortDistribution.cs:60]: I have added a stub function called `RetrieveParticipantData`

File: application/CohortManager/src/Functions/CohortDistributionServices/CreateCohortDistribution/CreateCohortDistribution.cs
Comment (andrewmorris43) [reply application/CohortManager/src/Functions/CohortDistributionServices/CreateCohortDistribution/CreateCohortDistribution.cs:1]: This has been refactored into methods that handle the separate tasks.

**Review by MWClayson-NHS — APPROVED** [review]
Looks good to me!

**Pull request merged**